src/main/java/org/testng/internal/Systematiser.java

```diff
@@ -0,0 +1,86 @@
+package org.testng.internal;
+
+import org.testng.ITestNGMethod;
+
+import java.util.Comparator;
+
+public class Systematiser {
+
+    private Systematiser() {
+        //Utility class. Defeat instantiation.
+    }
+
+    public static Comparator<Graph.Node> getComparator() {
```

Comment:
I don't catch it because `o1.getObject().toString()` will work with `Node<ITestNGMethod>` too.